spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/transaction/jta/NarayanaJtaConfiguration.java

```diff
@@ -114,10 +115,12 @@ public TransactionManager narayanaTransactionManager() {
 	@Bean
 	@DependsOn("narayanaConfiguration")
 	public RecoveryManagerService narayanaRecoveryManagerService() {
+		RecoveryManager.delayRecoveryManagerThread();
```

Comment:
This method sets a boolean to `true`. It makes sure that recovery manager has to be started explicitly. Without it recovery manager is started automatically when any transaction logic is executed for the first time. This is one of the candidates to be moved out as per our discussion. But for now it makes sense here.
As for integration tests, I'm not aware of any. Do you have any integration tests which I should check?